pkg/mirrorregistries/cluster_image_registry.go

```diff
@@ -0,0 +1,228 @@
+}
+
+// parseMirrorRegistries takes a mirror registry toml tree and parses it into
+// a list of image mirrors and a string list of insecure mirror registries.
+func parseMirrorRegistries(
+	idmsMirrors *[]configv1.ImageDigestMirrors,
+	itmsMirrors *[]configv1.ImageTagMirrors,
+	mirrorTrees []*toml.Tree,
+	insecureRegistries *[]string,
+	source string,
+) {
+	itmsMirror := configv1.ImageTagMirrors{}
+	idmsMirror := configv1.ImageDigestMirrors{}
+	for _, mirrorTree := range mirrorTrees {
+		if mirror, ok := mirrorTree.Get("location").(string); ok {
+			if insecure, ok := mirrorTree.Get("insecure").(bool); ok && insecure {
+				*insecureRegistries = append(*insecureRegistries, mirror)
+			}
+			if pullFrom, ok := mirrorTree.Get("pull-from-mirror").(string); ok {
+				switch pullFrom {
+				case "tag-only":
+					itmsMirror.Source = source
+					itmsMirror.Mirrors = append(itmsMirror.Mirrors, configv1.ImageMirror(mirror))
+				case "digest-only":
+					idmsMirror.Source = source
+					idmsMirror.Mirrors = append(idmsMirror.Mirrors, configv1.ImageMirror(mirror))
+				}
+			} else {
+				// Default is pulling by digest
+				idmsMirror.Source = source
+				idmsMirror.Mirrors = append(idmsMirror.Mirrors, configv1.ImageMirror(mirror))
+			}
+		}
+	}
+	if len(itmsMirror.Mirrors) > 0 {
+		*itmsMirrors = append(*itmsMirrors, itmsMirror)
+	}
+	if len(idmsMirror.Mirrors) > 0 {
+		*idmsMirrors = append(*idmsMirrors, idmsMirror)
+	}
+}
+
+// processMirrorRegistryConfig retrieves the mirror registry configuration from registries.conf and ca-bundle.crt
+func processMirrorRegistryConfig(registriesConf, caBundleCrt string) (*hiveext.MirrorRegistryConfiguration, error) {
+	if registriesConf == "" {
+		return nil, nil
+	}
+
+	imageDigestMirrors, imageTagMirrors, insecure, err := GetImageRegistries(registriesConf)
+	if err != nil {
+		return nil, err
+	}
+
+	// Store the mirror configuration for later use during install config generation
+	mirrorRegistryConfigurationInfo := &hiveext.MirrorRegistryConfigurationInfo{
+		ImageDigestMirrors: imageDigestMirrors,
+		ImageTagMirrors:    imageTagMirrors,
+		Insecure:           insecure,
+	}
+
+	return &hiveext.MirrorRegistryConfiguration{
+		MirrorRegistryConfigurationInfo: mirrorRegistryConfigurationInfo,
+		CaBundleCrt:                     caBundleCrt,
+		RegistriesConf:                  registriesConf,
+	}, nil
+}
+
+// getUserTomlConfigMapData get registries.conf and ca-bundle.crt if exist in the provided configmap inside AgentClusterInstall
+func getUserTomlConfigMapData(ctx context.Context, log logrus.FieldLogger, c client.Client, ref *hiveext.MirrorRegistryConfigMapReference) (string, string, error) {
+	userTomlConfigMap := &corev1.ConfigMap{}
+	err := c.Get(ctx, types.NamespacedName{Name: ref.Name, Namespace: ref.Namespace}, userTomlConfigMap)
+	if err != nil {
+		log.Error(err, "Failed to get ConfigMap", "ConfigMapName", ref.Name, "ConfigMapNamespace", ref.Namespace)
```

Comment:
do we have a validation for a missing CM? probably need some condition error in status for that.

Reply:
We update the status [here](https://github.com/openshift/assisted-service/pull/6646/files#diff-40883156aab921707c5a73dfa3b451bbd3accb4c0636d715d0a7768db8904b57R208). Is that enough?